lua/refactoring/init.lua

```diff
@@ -2,9 +2,10 @@ local refactors = require("refactoring.refactor")
 local Config = require("refactoring.config")
 
 local M = {}
+local config = {}
```

Comment:
Let me think about this.  

Does this mean everywhere else that is calling the config is getting the default config?  Do we need to do the threading through starting here?

Reply:
Well actually, not as of right now. Since `config` is local, only functions which will be passed the variable will use it.

Reply:
I like the idea of that, but I don't know if every refactor should have its own config? Thinking widely, you configure and setup your plugin and then, your Config is system wide? Or am I again missing the point?

Why would configuring every call to the plugin separately useful? What would differ?

Then, as I said, I agree with the singleton, probably paired with a sort of "Prototype" should be the way to go.

Reply:
Also, since oop isn't really baked into Lua, would adding a dependency worth the trouble?